### PR TITLE
Fix S2190 FN: Support indexer, conversion operator and event declaration

### LINKED ISSUE #6642: FN S2190: No issues raised if recursion is inside an indexer
### Description

No issues are raised if recursion is inside an indexer.

### Repro steps

```
class SampleCollection
{
    public int this[int i]
    {
        get { return this[i]; } // FN
        set { this[i] = value; } // FN
    }
}
```

### Expected behavior

Issues should be raised if recursion is inside an indexer.

### Actual behavior

No issues are raised if recursion is inside an indexer.

### Known workarounds

-

### Related information

* C#/VB.NET Plugins version: 8.51.0.59060

### LINKED ISSUE #6643: FN S2190: No issues raised if recursion is inside a ConversionOperatorDeclaration
### Description

No issues are raised if recursion is inside a ConversionOperatorDeclaration.

### Repro steps

```
public class Digit
{
    public static implicit operator byte(Digit d) => d;
}
```

### Expected behavior

Issues should be raised if recursion is inside a ConversionOperatorDeclaration.

### Actual behavior

No issues are raised if recursion is inside a ConversionOperatorDeclaration.

### Known workarounds

-

### Related information

* C#/VB.NET Plugins version: 8.51.0.59060

### LINKED ISSUE #6644: FN S2190: No issues raised if recursion is inside an EventDeclaration
### Description

No issues are raised if recursion is inside an EventDeclaration.

### Repro steps

```
public delegate bool SomeDelegate();

public class SomeClass
{
    public event SomeDelegate OtherEvent
    {
        add
        {
            OtherEvent += value;
        }

        remove
        {
            OtherEvent -= value;
        }
    }
}
```

### Expected behavior

Issues should be raised if recursion is inside an EventDeclaration.

### Actual behavior

No issues are raised if recursion is inside an EventDeclaration.

### Known workarounds

-

### Related information

* C#/VB.NET Plugins version: 8.51.0.59060

### PR DESCRIPTION
**Pull request by sagi1623**
Fixes #6642 
Fixes #6643 
Fixes #6644

### CODE REVIEW

**Comment by sagi1623** [comment]
I fixed the code smells and opened an [issue](https://github.com/SonarSource/sonar-dotnet/issues/8502) for the rest of the FPs that remain in the file. Regarding the coverage I could not come up with any valid or invalid code snippet that would cover the cases, however I would rather leave it there as defensive coding than to create AD0001s if anything changes in Roslyn.

**Review by cristian-ambrosini-sonarsource — CHANGES_REQUESTED** [review]
Left few ideas to bump the coverage

File: analyzers/src/SonarAnalyzer.CSharp/Rules/InfiniteRecursion.SonarCfg.cs [hunk 1]
```diff
@@ -60,6 +60,16 @@ public void CheckForNoExitProperty(SonarSyntaxNodeReportingContext c, PropertyDe
                 }
             }
 
+            public void CheckForNoExitIndexer(SonarSyntaxNodeReportingContext c, IndexerDeclarationSyntax indexer, IPropertySymbol propertySymbol)
+            {
+                // SonarCFG is out of support
+            }
```
[the remark this answers is not in the document]
Comment (mary-georgiou-sonarsource) [reply analyzers/src/SonarAnalyzer.CSharp/Rules/InfiniteRecursion.SonarCfg.cs:66]: We don't support and we don't care about the sonar SE engine. It's deprecated.

**Review by CristianAmbrosini — APPROVED** [review]
LGTM! Feel free to rebase and merge

**Comment by CristianAmbrosini** [comment]
@sagi1623 Freedom! Finally 😄

**Pull request merged**